toc/rfc/rfc-draft-ipv6-dual-stack-for-cf.md

```diff
@@ -0,0 +1,200 @@
+2. When publicly routable addresses (Global Unicast Addresses (GUAs) in IPv6) are assigned to each application container, traffic from outside the foundation needs to be blocked.
+
+BBS, Executor and Garden components need to be extended to support the ICMPv6 protocol for ASGs.
+
+Finally, some adaptations for the containerized workloads are necessary:
+
+1. The [environment variables](https://docs.cloudfoundry.org/devguide/deploy-apps/environment-variable.html#app-system-env) `CF_INSTANCE_IP` and `CF_INSTANCE_INTERNAL_IP` will need IPv6 equivalents, or extension.
+2. Instance identity IDs, i.e. the certificates used by Envoy in the app container, need to support IPv6 addresses. Currently they are issued for the IPv4 address only.
+
+### silk-release
+
+The CNI plugin needs to be extended to support IPv6 address assignment. Currently, container-facing interfaces are created and assigned an IPv4 address using the host-local IPAM plugin. The required enhancement involves assigning an IPv6 address alongside IPv4 and updating the state files accordingly.
+
+The CNI wrapper plugin must be extended to support IPv6. Currently, it creates default iptables rules (`netin-*`, `netout-*` chains) with default `REJECT` rules and `ALLOW` rules for DNS and "host services." In deployments without vxlan-policy-agent enabled, it also manages iptables egress rules. The enhancement requires adding logic to handle IPv6 rules using ip6tables.
+
+Additionally, the VXLAN Policy Agent must be updated to support IPv6 Address Security Groups (ASGs) received from the Policy Server.
+
+### Cloud Controller
+
+Changes to the cloud controller are primarily in the following areas:
+
+1. Support forwarding and retrieving of IPv6 addresses in any records that may have IP addresses
+2. Support binding to IPv4 and IPv6 for the CAPI server, depending on the VM’s configuration
+3. Support of IPv6 in metadata, e.g. Application Security Groups
+
+### Cloud Foundry CLI
+
+The Cloud Foundry CLI shows the application workload’s addresses in various places. Those places need to be extended to support the display of IPv6 addresses, as retrieved via the CF API:
+
+1. IPv6 support for configuring application security groups
+2. …
+
+### Networking
+
+Network Policies that are stored in [the policy-server-api which is part of cf-networking-release](https://github.com/cloudfoundry/cf-networking-release/blob/develop/docs/08-policy-server-api.md).
+
+Network policies may reference IP address ranges as destination. Destinations are stored as strings. The underlying IP ranges are also [strings with a start and end address](https://github.com/cloudfoundry/cf-networking-release/blob/aa25a3f96dc21e8a3e25db10f741e4f678c78464/src/code.cloudfoundry.org/policy-server/api/api.go#L37-L49). The data is ultimately serialized JSON that is stored in a database column. As IPv6 addresses are longer than IPv4 addresses, we might need to bump the table size.
+
```

Comment:
Same here with the dual-stack. If we want to roll this out for existing deployments dual-stack should be supported for backwards compatibility. 

Most of the components only set up IP address right now.

Reply:
I will add some sentence in the beginning that highlights that this is about adding IPv6 and thus reaching dual stack.